dev-scripts/prepare-chart

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -e
+
+DEV_SCRIPT_DIR=$(dirname "$(realpath "${BASH_SOURCE[0]}")")
+SCRIPT_DIR=$(realpath "$DEV_SCRIPT_DIR/../scripts")
+
+echo "Dev-Script: Prepare Rancher Chart"
+
+source "$SCRIPT_DIR/version"
+
+test_image="$REPO/rancher"
+
+echo "Preparing Rancher Chart for release: ${CHART_VERSION}"
+
+sed -i -e "s/%VERSION%/${CHART_VERSION}/g" ./chart/Chart.yaml
```

Comment:
On macOS the `-i` argument in `sed` requires an extension separated by whitespace after `-i`. On linux the extension is optional and needs to be attached to the `-i`, as in `sed -i""` for linux, and `sed -i ""` for macOS

If this script can be used outside dapper I would recommend finding a 2-platform solution. Maybe use a temp file and drop `-i` completely.

Also you can have multiple `-e` transforms, so why not:

```
set -e
set -o pipefail
...
TMP=$(mktemp)
sed -e "s/%APP_VERSION%/${APP_VERSION}/g" ./chart/Chart.yaml > $TMP && mv $TMP ./chart/Chart.yaml
rm "$TMP"
TMP=$(mktemp)
sed -e "s@%POST_DELETE_IMAGE_NAME%@${test_image}@g" \
   -e "s/%POST_DELETE_IMAGE_TAG%/${APP_VERSION}/g" \
   ./chart/values.yaml > $TMP && mv $TMP ./chart/values.yaml
rm "$TMP"
```

Reply:
I've added a workaround we (o&b squad) used within `prometheus-federator` build scripts to solve the `sed` incompatibilities. This is intended to be used without dapper for sure.